Xamarin.Forms.Platform.Android/AppCompat/TabbedPageRenderer.cs

```diff
@@ -407,55 +632,329 @@ void UpdateTabIcons()
+					{
+						if (_backgroundDrawable == null)
+							_backgroundDrawable = _bottomNavigationView.Background;
+						_bottomNavigationView.SetBackgroundColor(tintColor.ToAndroid());
+					}
+				}
+			}
+			else
 			{
-				if (tintColor.IsDefault)
-					_tabLayout.BackgroundTintMode = null;
+				Color tintColor = Element.BarBackgroundColor;
+
+				if (Forms.IsLollipopOrNewer)
+				{
+					if (tintColor.IsDefault)
+						_tabLayout.BackgroundTintMode = null;
+					else
+					{
+						_tabLayout.BackgroundTintMode = PorterDuff.Mode.Src;
+						_tabLayout.BackgroundTintList = ColorStateList.ValueOf(tintColor.ToAndroid());
+					}
+				}
 				else
 				{
-					_tabLayout.BackgroundTintMode = PorterDuff.Mode.Src;
-					_tabLayout.BackgroundTintList = ColorStateList.ValueOf(tintColor.ToAndroid());
+					if (tintColor.IsDefault && _backgroundDrawable != null)
+						_tabLayout.SetBackground(_backgroundDrawable);
+					else if (!tintColor.IsDefault)
+					{
+						if (_backgroundDrawable == null)
+							_backgroundDrawable = _tabLayout.Background;
+						_tabLayout.SetBackgroundColor(tintColor.ToAndroid());
+					}
 				}
 			}
+		}
+
+		protected virtual ColorStateList GetItemTextColorStates()
+		{
+			if (IsDisposed)
+				return null;
+
+			if (IsBottomTabPlacement)
+			{
+				if (_originalTabTextColors == null)
+					_originalTabTextColors = _bottomNavigationView.ItemTextColor;
+			}
 			else
 			{
-				if (tintColor.IsDefault && _backgroundDrawable != null)
-					_tabLayout.SetBackground(_backgroundDrawable);
-				else if (!tintColor.IsDefault)
+				if (_originalTabTextColors == null)
+					_originalTabTextColors = _tabLayout.TabTextColors;
+			}
+
+			Color barItemColor = BarItemColor;
+			Color barTextColor = Element.BarTextColor;
+			Color barSelectedItemColor = BarSelectedItemColor;
+
+			if (barItemColor.IsDefault && barTextColor.IsDefault && barSelectedItemColor.IsDefault)
+				return _originalTabTextColors;
+
+			if (_newTabTextColors != null)
+				return _newTabTextColors;
+
+			int checkedColor;
+			int defaultColor;
+
+			if (!barTextColor.IsDefault)
+			{
+				checkedColor = barTextColor.ToAndroid().ToArgb();
+				defaultColor = checkedColor;
+			}
+			else
+			{
+				defaultColor = barItemColor.ToAndroid().ToArgb();
+
+				if (barItemColor.IsDefault && _originalTabTextColors != null)
+				{
+					defaultColor = _originalTabTextColors.DefaultColor;
+				}
+
+				checkedColor = defaultColor;
+
+				if (!barSelectedItemColor.IsDefault)
 				{
-					if (_backgroundDrawable == null)
-						_backgroundDrawable = _tabLayout.Background;
-					_tabLayout.SetBackgroundColor(tintColor.ToAndroid());
+					checkedColor = barSelectedItemColor.ToAndroid().ToArgb();
+				}
+			}
+
+			_newTabTextColors = GetColorStateList(defaultColor, checkedColor);
+			return _newTabTextColors;
+		}
+
+		protected virtual ColorStateList GetItemIconTintColorState()
+		{
+			if (IsDisposed)
+				return null;
+
+			if (IsBottomTabPlacement)
+			{
```

Comment:
again 1 line I would drop the bracket

Reply:
I dropped a bunch of brackets but this one I think is fine due to the if statement below this. So it's kind of the same conditional flow as here

https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.Android/AppCompat/TabbedPageRenderer.cs#L211